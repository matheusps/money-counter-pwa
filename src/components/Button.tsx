import styled from 'styled-components'

const Button = styled.button`
  background-color: ${props => props.theme.light};
  color: ${props => props.theme.dark};
  width: 100%;
  font-size: 1.5rem;
  padding: 0.8rem 2.6rem 0.8rem 2.6rem;
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-weight: 500;
  font-stretch: normal;
  line-height: 1.4;
  margin: 0.2rem;
  display: relative;
  overflow: hidden;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  transition: transform 0.2s ease-in;
  will-change: transform;
  :hover {
    transform: scale(0.975);
    -webkit-transform: scale(0.975);
  }
  :active {
    transform: scale(0.9);
    -webkit-transform: scale(0.9);
  }
  :focus {
    outline: none;
  }
`

export default Button
