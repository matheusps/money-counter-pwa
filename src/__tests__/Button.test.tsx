import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'

import Button from '../components/Button'

describe('Button component', () => {
  const component = render(<Button>Testing</Button>)

  it('should render', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
