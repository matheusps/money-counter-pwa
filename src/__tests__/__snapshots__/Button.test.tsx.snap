// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  font-size: 1.5rem;
  padding: 0.8rem 2.6rem 0.8rem 2.6rem;
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-weight: 500;
  font-stretch: normal;
  line-height: 1.4;
  margin: 0.2rem;
  display: relative;
  overflow: hidden;
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: -webkit-transform 0.2s ease-in;
  -webkit-transition: transform 0.2s ease-in;
  transition: transform 0.2s ease-in;
  will-change: transform;
}

.c0:hover {
  -webkit-transform: scale(0.975);
  -ms-transform: scale(0.975);
  transform: scale(0.975);
  -webkit-transform: scale(0.975);
}

.c0:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
  -webkit-transform: scale(0.9);
}

.c0:focus {
  outline: none;
}

<button
    class="c0"
  >
    Testing
  </button>
</DocumentFragment>
`;
